### PR TITLE
fix: Projects Service Types

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 4a62a4f</samp>

Added or improved type annotations for the `ProjectsService` class and its test data. This improved the type safety, readability, and consistency of the code and the tests.

### <samp>🤖 Generated by Copilot at 4a62a4f</samp>

> _Oh we are the coders of the `ProjectsService` class_
> _We annotate our types to make our code more fast_
> _We heave and we ho on the count of three_
> _We import our models and we use `Date` for `created_at` and `updated_at`_

### <samp>🤖 Generated by Copilot at 4a62a4f</samp>

*  Import `OrgCategory` model and use it as a type for `activeCategoryList` and `categoryList` in `ProjectsService` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-e792da4f87c1f0363cc743e99c96daed35d1480c037f0f893f9529090a40a5ddL11-R23), [link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-e792da4f87c1f0363cc743e99c96daed35d1480c037f0f893f9529090a40a5ddL113-R124), [link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-4ee784932d39789c2fa636ec3a9d37789c634fe10de18e64605fa0896865b24fR2))
* Define and use `ProjectParamsUnformatted` type for `projectParams` in `getByParamsUnformatted` method of `ProjectsService` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-e792da4f87c1f0363cc743e99c96daed35d1480c037f0f893f9529090a40a5ddL19-R31))
* Remove `any` type annotation from `params` in `getByParamsUnformatted` method of `ProjectsService` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-e792da4f87c1f0363cc743e99c96daed35d1480c037f0f893f9529090a40a5ddL38-R38))
* Add `as number[]` type assertion to `orgCategoryIds` in `addOrgCategoryIdsFilter` method call in `getByParamsUnformatted` method of `ProjectsService` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-e792da4f87c1f0363cc743e99c96daed35d1480c037f0f893f9529090a40a5ddL49-R49))
* Specify generic type parameters for `get` method calls of `apiV2Service` in `getByParamsUnformatted` and `getbyId` methods of `ProjectsService` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-e792da4f87c1f0363cc743e99c96daed35d1480c037f0f893f9529090a40a5ddL58-R68), [link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-e792da4f87c1f0363cc743e99c96daed35d1480c037f0f893f9529090a40a5ddL147-R157))
* Convert `created_at` and `updated_at` properties to `Date` objects in test data in `projects.spec.data.ts` file ([link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-4ee784932d39789c2fa636ec3a9d37789c634fe10de18e64605fa0896865b24fL172-R173), [link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-4ee784932d39789c2fa636ec3a9d37789c634fe10de18e64605fa0896865b24fL180-R185), [link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-4ee784932d39789c2fa636ec3a9d37789c634fe10de18e64605fa0896865b24fL192-R197), [link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-4ee784932d39789c2fa636ec3a9d37789c634fe10de18e64605fa0896865b24fL204-R205), [link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-4ee784932d39789c2fa636ec3a9d37789c634fe10de18e64605fa0896865b24fL211-R212), [link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-4ee784932d39789c2fa636ec3a9d37789c634fe10de18e64605fa0896865b24fL219-R224), [link](https://github.com/fylein/fyle-mobile-app/pull/2100/files?diff=unified&w=0#diff-4ee784932d39789c2fa636ec3a9d37789c634fe10de18e64605fa0896865b24fL231-R232))

## Clickup
https://app.clickup.com/t/85zt699np

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/services/projects.service.ts [hunk 1]
```diff
@@ -8,34 +8,34 @@ import { ExtendedProject } from '../models/v2/extended-project.model';
 import { ProjectV1 } from '../models/v1/extended-project.model';
 import { ProjectParams } from '../models/project-params.model';
 import { intersection } from 'lodash';
-
+import { OrgCategory } from '../models/v1/org-category.model';
+
+type ProjectParamsUnformatted = Partial<{
```
Comment (yksolanki9) [inline src/app/core/services/projects.service.ts:13]: Dont use partial when declaring types. Use it when assigning type to a variable.
Also, move this to a model file

File: src/app/core/test-data/projects.spec.data.ts [hunk 2]
```diff
@@ -169,66 +170,66 @@ export const apiV2ResponseSingle = {
 export const testActiveCategoryList = [
   {
     code: '4060340',
-    created_at: '2018-01-31T23:50:27.215171+00:00',
+    created_at: new Date('2018-01-31T23:50:27.215171+00:00'),
```
Comment (yksolanki9) [inline src/app/core/test-data/projects.spec.data.ts:173]: The api returns a strring right? We shouldnt be converting it to Date object. Change the type of created_at to string

File: src/app/core/services/projects.service.ts [hunk 3]
```diff
@@ -8,34 +8,34 @@ import { ExtendedProject } from '../models/v2/extended-project.model';
...
+  orgId: string;
+  active: boolean;
+  orgCategoryIds: number[] | string[];
```
Comment (yksolanki9) [inline src/app/core/services/projects.service.ts:16]: orgCategoryIds are always numbers iirc.

File: src/app/core/services/projects.service.ts [hunk 4]
```diff
@@ -46,7 +46,7 @@ export class ProjectsService {
     this.addActiveFilter(active, params);
 
     // `orgCategoryIds` can be optional
-    this.addOrgCategoryIdsFilter(orgCategoryIds, params);
+    this.addOrgCategoryIdsFilter(orgCategoryIds as number[], params);
```
Comment (yksolanki9) [inline src/app/core/services/projects.service.ts:49]: If we remove the union this will not be needed

**Review by yksolanki9 — CHANGES_REQUESTED** [review]

**Review by Dimple16 — COMMENTED** [review]
already reviewed, can skip my review

**Pull request closed without merging**